UOP1_Project/Assets/Scripts/Audio/SoundEmitters/SoundEmitter.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+using UOP1.Pool;
+using System;
+using UnityEngine.Events;
+
+[RequireComponent (typeof(AudioSource))]
+public class SoundEmitter : MonoBehaviour, IPoolable
+{
+	private AudioSource _audioSource;
+	private float _lastUseTimestamp = 0;
+
+	public event UnityAction<SoundEmitter> OnSoundFinishedPlaying;
+
+	private void Awake()
+	{
+		_audioSource = this.GetComponent<AudioSource>();
+		_audioSource.playOnAwake = false;
+	}
+
+	public void OnRequest() { }
+
+	public void OnReturn(Action onReturned)
+	{
+		StopSound();
```

Comment:
With the current pool implementation, SoundEmitters will never be returned to the pool if they do not invoke onReturned. PoolableParticle in the example files shows how to do this.

Note: PR #159 makes it so resetting is entirely the responsibility of the pooled object or its manager and does not require any callback. This code will still need to be adjusted in the case that PR is merged.